js/flightlog.js

```diff
@@ -296,6 +296,35 @@ function FlightLog(logData) {
         return numMotors;
     };
 
+    /*
+     * Propagates array data fiels over empty fields
+     * Debug fields can be arrays with data, so it fills data gaps when no arrays are stored
+     */
+    function propagateArrayFields(chunks)
+    {
+        var k, j, i, previousFrame = 0, currentFrame;
+        
+        for (k = 0; k < chunks.length; k++)
+        {
+            for (j = 0; j < chunks[k].frames.length; j++)
+            {
+                currentFrame = chunks[k].frames[j]; 
+                if (previousFrame == 0)
+                    previousFrame = currentFrame;
+                    
+                for (i = 0; i < currentFrame.length; i++)
+                {
+                    if (!Array.isArray(currentFrame[i]) && Array.isArray(previousFrame[i]))
+                    {
+                        currentFrame[i] = previousFrame[i];
+                    }
+                }
+                
+                previousFrame = chunks[k].frames[j];
+            }
+        }
+    }
```

Comment:
Please try to avoid using var in new code. Braces are a part of the thought containing the conditional or loop. Therefore, they belong on the same line.
```suggestion
        let previousFrame = 0, currentFrame;
        
        for (let k = 0; k < chunks.length; k++) {
            for let (j = 0; j < chunks[k].frames.length; j++) {
                currentFrame = chunks[k].frames[j]; 
                if (previousFrame === 0)
                    previousFrame = currentFrame;
                    
                for (let i = 0; i < currentFrame.length; i++) {
                    if (!Array.isArray(currentFrame[i]) && Array.isArray(previousFrame[i])) {
                        currentFrame[i] = previousFrame[i];
                    }
                }
                
                previousFrame = chunks[k].frames[j];
            }
        }
    }
```

Reply:
refactored